domains/transfer-agreements/TransferAgreements.API/API/Data/ITransferAgreementRepository.cs

```diff
@@ -1,8 +1,12 @@
+using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
+using API.ApiModels.Responses;
 
 namespace API.Data;
 
 public interface ITransferAgreementRepository
 {
     Task<TransferAgreement> AddTransferAgreementToDb(TransferAgreement transferAgreement);
+    Task<List<TransferAgreementResponse>> GetTransferAgreementsBySubjectId(Guid subjectId);
```

Comment:
I dont think the repository should be aware of the response DTO it could return `List<TransferAgreement>`